resources/assets/coffee/react/_components/countdown-timer.coffee

```diff
@@ -0,0 +1,66 @@
+###
+# Copyright 2015-2016 ppy Pty. Ltd.
+#
+# This file is part of osu!web. osu!web is distributed with the hope of
+# attracting more community contributions to the core ecosystem of osu!.
+#
+# osu!web is free software: you can redistribute it and/or modify
+# it under the terms of the Affero GNU General Public License version 3
+# as published by the Free Software Foundation.
+#
+# osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+{div} = React.DOM
+el = React.createElement
+
+bn = 'countdown-timer'
+
+class @CountdownTimer extends React.Component
+  constructor: (props) ->
+    super props
+    @state =
+      deadline: @props.deadline
```

Comment:
looks like parsing it once (`moment(@props.deadline)`) and calling the diff repeatedly is enough.